### PR TITLE
chore: archive migrations 2023 or older and test if all migrations ran

### PR DESCRIPTION
**Pull request by hejtful**
resolves https://github.com/serlo/db-migrations/issues/287
resolves https://github.com/serlo/db-migrations/issues/292

### CODE REVIEW

File: .github/actions/test-migrations/action.yml [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
   using: composite
   steps:
-    # With the following command we check that the migration
-    # "20220625202000-convert-important-and-blockquote-to-box" was run
-    # successfully
-    - run:
-        yarn mysql:list-migrations:ci | grep
-        20220625202000-convert-important-and-blockquote-to-box
+    - run: |
+        chmod +x ./scripts/check_all_migrations_run.sh
```
Comment (kulla) [inline .github/actions/test-migrations/action.yml:9]: When you run `chmod +x ./scripts/check_all_migrations_run.sh` locally you should be able to push this change to the git repo.

File: src/utils/edtr-io.ts [hunk 2]
```diff
@@ -1,117 +1,5 @@
...
-    }
-
-    return value
-  }
-
-  return applyTransformation
-}
-
-type ListTransformation<A> = (element: A) => A[] | undefined
-
```
Comment (kulla) [inline src/utils/edtr-io.ts:114]: Can you leave them here -> We will need this in the future...

**Review by kulla — CHANGES_REQUESTED** [review]

**Comment by kulla** [comment]
@hejtful Can you run the archived db-migrations in https://github.com/serlo/serlo-mysql-database?tab=readme-ov-file#change-db-via-migrations? -> They might change the DB structure which should be reflected here...

File: src/utils/edtr-io.ts (hunk 2 above)
Comment (hejtful) [reply src/utils/edtr-io.ts]: Sure, good point. I opened an issue to rename edtr-io to serlo-editor, though: https://github.com/serlo/db-migrations/issues/298

**Comment by hejtful** [comment]
@kulla I thought that the point of archiving the migrations was to prevent them from being run. I'm not sure what you want me to do now. Should I somehow run the archived migrations from this (`chore/archive-migrations-2023-or-older`) branch? Or do you want me to run them from the `main` branch? And what is the result that I'm aiming for? If the DB structure is unchanged, then the migrations can safely be archived?

**Review by kulla — APPROVED** [review]
@hejtful :heart:

**Pull request merged**